4.gene_expression_signatures/scripts/1.download-validation-data.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Download and Process Neuroblastoma RNAseq Data
+
+# Code from: https://github.com/greenelab/BioBombe/blob/master/10.gene-expression-signatures/0.download-validation-data.ipynb
+# 
+# We are downloading the dataset associated with Harenza et al. 2017. The data profiles RNAseq data from 39 commonly used neuroblastoma (NBL) cell lines.
+# 
+# We are interested in the MYCN amplification status of these cell lines. We will test if the MYCN amplification score learned through the BioBombe signature approach applied to TARGET data generalizes to this cell line dataset.
+# 
+# MYCN Amplification refers to the number of copies of the MYCN gene. MYCN amplification is used as a biomarker for poor prognosis in neuroblastoma patients (Huang and Weiss 2013).
+
+# In[1]:
+
+
+import os
+import requests
+import pandas as pd
+from urllib.request import urlretrieve
+
+from sklearn import preprocessing
+
+
+# In[2]:
+
+
+url = "https://ndownloader.figshare.com/files/14138792"
+name = "2019-01-22-CellLineSTAR-fpkm-2pass_matrix.txt"
+path = os.path.join("data", name)
+
+
+# In[3]:
+
+
+os.makedirs("data", exist_ok=True)
+
+
+# In[4]:
+
+
+urlretrieve(url, path)
+
+
+# In[5]:
+
+
+get_ipython().system(' md5sum "data/2019-01-22-CellLineSTAR-fpkm-2pass_matrix.txt"')
+
+
+# # Download Phenotype Data
+
+# In[6]:
+
+
+url = "https://www.nature.com/articles/sdata201733/tables/4"
+name = "nbl_cellline_phenotype.txt"
+path = os.path.join("data", name)
+
+
+# In[7]:
+
+
+if not os.path.isfile(path):
+    html = requests.get(url).content
+
+    pheno_df = pd.read_html(html)[0]
+    pheno_df['Cell Line'] = pheno_df['Cell Line'].str.replace("-", "")
+
+    pheno_df.to_csv(path, sep='\t', index=False)
+
+else:
+    pheno_df = pd.read_csv(path, sep="\t")
+
+pheno_df.head()
+
+
+# In[8]:
+
+
+get_ipython().system(' md5sum "data/nbl_cellline_phenotype.txt"')
+
+
+# # Process RNAseq Data
+
+# In[9]:
+
+
+raw_file = os.path.join("data", "2019-01-22-CellLineSTAR-fpkm-2pass_matrix.txt")
+
+raw_df = pd.read_table(raw_file, sep='\t')
+raw_df.head()
+
+
+# # Update Gene Names
+
+# In[10]:
+
+
+# Load curated gene names from versioned resource 
+commit = '721204091a96e55de6dcad165d6d8265e67e2a48'
+url = 'https://raw.githubusercontent.com/cognoma/genes/{}/data/genes.tsv'.format(commit)
+gene_df = pd.read_table(url)
+
+# Only consider protein-coding genes
+gene_df = (
+    gene_df.query("gene_type == 'protein-coding'")
+)
+
+symbol_to_entrez = dict(zip(gene_df.symbol,
+                            gene_df.entrez_gene_id))
+
```

Comment:
Why? LNC-RNAs are interesting too

Reply:
Currently consistency with original graph, but this could be changed in future

Reply:
Aha, I see. This would be an interesting analysis to look at in the future!